src/trace/implementations/spine_fueled.rs

```diff
@@ -483,7 +490,7 @@ where
         }
 
         // Having performed all of our work, if more than one batch remains reschedule ourself.
-        if !self.reduced() {
+        if !self.exert_effort().is_some() {
```

Comment:
```suggestion
        if self.exert_effort().is_none() {
```

Reply:
Ooh, I think this is wrong both ways, actually. We want to reschedule if there is exert effort to apply. Oops. Will fix.